### ISSUE TITLE
Null reference exception when calling BulkSaveChanges under EF 6.5.1 with Microsoft.Data.SqlClient provider.

### ISSUE THREAD

**Issue by ScruffyMagicM** [NorthernLight1/N.EntityFramework.Extensions#138]
Describe the bug
I'm trying to call BulkSaveChanges, but I'm receiving this error: Object reference not set to an instance of an object.

Additional information:
I'm using entity framework 6.5.1 with Microsoft.Data.SqlClient as the provider. The same code worked without issue (6.4.4, System.Data.SqlClient) before I migrated to the newer entity framework version and provider, so I suspect it's relevant. The library describes says it supports 6.4.4+, which I took to mean 6.5.1 was included.

Stack Trace:

   at N.EntityFramework.Extensions.DbTransactionContext..ctor(DbContext context, ConnectionBehavior connectionBehavior, TransactionalBehavior transactionalBehavior, Nullable`1 commandTimeout, Boolean openConnection)

Call Stack:

```
N.EntityFramework.Extensions.dll!N.EntityFramework.Extensions.DbTransactionContext.DbTransactionContext(System.Data.Entity.DbContext context, N.EntityFramework.Extensions.Enums.ConnectionBehavior connectionBehavior, System.Data.Entity.TransactionalBehavior transactionalBehavior, int? commandTimeout, bool openConnection)	Unknown

N.EntityFramework.Extensions.dll!N.EntityFramework.Extensions.DbTransactionContext.DbTransactionContext(System.Data.Entity.DbContext context, N.EntityFramework.Extensions.BulkOptions options)	Unknown
 	N.EntityFramework.Extensions.dll!N.EntityFramework.Extensions.DbContextExtensions.BulkInsert<object>(System.Data.Entity.DbContext context, System.Collections.Generic.IEnumerable<object> entities, N.EntityFramework.Extensions.BulkInsertOptions<object> options)	Unknown
 	N.EntityFramework.Extensions.dll!N.EntityFramework.Extensions.DbContextExtensions.BulkInsert<object>(System.Data.Entity.DbContext context, System.Collections.Generic.IEnumerable<object> entities, System.Action<N.EntityFramework.Extensions.BulkInsertOptions<object>> optionsAction)	Unknown
 	N.EntityFramework.Extensions.dll!N.EntityFramework.Extensions.DbContextExtensions.BulkSaveChanges(System.Data.Entity.DbContext dbContext, bool acceptAllChangesOnSuccess, bool autoMapOutput)	Unknown
 	N.EntityFramework.Extensions.dll!N.EntityFramework.Extensions.DbContextExtensions.BulkSaveChanges(System.Data.Entity.DbContext dbContext)	Unknown
```

To Reproduce

Call AddRange method.
Call BulkSaveChanges method.

My code is explicitly this:

```
myContext.MyDbSetEntitites.AddRange(newEntities);
myContext.BulkSaveChanges();
```

Expected behavior
Correct handling of null values.

Desktop (please complete the following information):

OS: Windows
Browser: Chrome
Version: 126.0.6478.127
Sql Sever: Microsoft SQL Server Express (64-bit) v16.0.1000.6

**Comment by SpicyCatGames** [comment]
Also happens with BulkInsert()
The actual exception is here in N.EntityFramework.Extensions.DbTransactionContext class
It seems `this.Connection` is null
![image](https://github.com/user-attachments/assets/d629adf0-b528-4623-bb19-27f017325d0b)

**Comment by penihel** [comment]
i have same problem here.

Same solution @NorthernLight1 ?

**Comment by penihel** [comment]
@NorthernLight1 , please check my sugestion

#142

**Comment by NorthernLight1** [comment]
The PR for this bug has been approved and released in v1.9.0.

**Comment by andkorsh** [comment]
I need this fix, but why v1.9.0 dropped support for .net framework 4.7.2? I currently use 472 and cannot upgrade at the moment.
